sysl2/sysl/eval/binexprEval.go

```diff
@@ -117,9 +115,26 @@ func (op DefaultBinExprStrategy) eval(txApp *sysl.Application, assign Scope, bin
 	if f, has := valueFunctions[key]; has {
 		return f(lhsValue, rhsValue)
 	}
+
 	panic(errors.Errorf("Unsupported operation:DefaultBinExprStrategy: %s", key))
 }
 
+func (op NegateBinExprStrategy) eval(txApp *sysl.Application, assign Scope, binexpr *sysl.Expr_BinExpr) *sysl.Value {
+	lhsValue := Eval(txApp, assign, binexpr.Lhs)
+	rhsValue := Eval(txApp, assign, binexpr.Rhs)
+
+	if binexpr.Op != sysl.Expr_BinExpr_NE {
+		panic(errors.Errorf("Attempting to call Negate Startergy on a non == operator"))
+	}
+
+	key := makeKey(sysl.Expr_BinExpr_EQ, getValueType(lhsValue), getValueType(rhsValue))
```

Comment:
Why not  replace the remainder of this function with `return unaryNeg(DefaultBinExprStrategy{}.eval(txApp, assign, binexpr))`?

Reply:
Oh yeah, thats much cleaner :D